ivy/data_classes/array/array.py

```diff
@@ -390,6 +390,8 @@ def __getattr__(self, item):
             attr = self._data.__getattribute__(item)
         except AttributeError:
             attr = self._data.__getattr__(item)
+        if callable(attr):
```

Comment:
I think the ideal behaviour should be not allowing the use of `permute` in the first place, if it wasn't present in `ivy`. @Infrared1029 what do you think?

Reply:
I agree, we should be adhering to our own API only. Ideally, this should fail (calling `permute`) without breaking the repo because I'm pretty sure this unintended behavior is sprinkled across the repo in some (could be many tbh) places.

Reply:
@vedpatwardhan @Infrared1029 that's what I thought too. But as you said, the '_feature_' seem to be used intentionally and unintentionally in the codebase, so it might break some stuff if we just naively prevent such method calls. So @Infrared1029 should we keep this quick fix for now and create another issue to look into this later ?

Reply:
hmm, I think if we keep this fix (allowing framework-specific methods to be called and returning an ivy.Array) it might reinforce the usage of such behavior more because the output is a valid ivy Array, my first time noticing this behavior is when I mistakenly used `ivy.permute` and noticed a `torch.Tensor` coming out.

thoughts @vedpatwardhan ?

Reply:
Hey @Infrared1029, yeah I think it would be much more clear to the user if they were never able to call framework-specific functions which didn't exist in `ivy` (e.g. `ivy.permute`) rather than it failing silently and them having to debug through a model with a bunch of functions to find that this function is deferring to a framework-specific function because of `getattr`. 
I agree with your point @kgmann, it might break some stuff but I think such usages should ideally break anyway. Ofc we wouldn't merge before fixing the issues arising from it. We can look through the PR workflows and fix all such failures if that's okay? 
Thanks both 🙂

Reply:
Sure @vedpatwardhan @Infrared1029 , and sorry for the delay. I just made some changes now. I'm not that familiar with the intelligent tests in the PR workflows yet, but I will look into them later and see if they are any new failures associated to the changes.